cypress-io/cypress
# chore(launchpad): support running e2e tests and ct tests in new launchpad

### Context

When we develop launchpad and app, we'll need e2e tests against a live gql server. This PR adds new commands to the package.json: 

**NOTE: Cypress Open + Cypress Run will execute against compiled versions of the Launchpad**

* `yarn cypress:open:e2e`
* `yarn cypress:open:ct`
* `yarn cypress:run:e2e`
* `yarn cypress:run:ct`
* `yarn cypress:run`

### Developing
No change when developing locally. `yarn dev` and `yarn start` are still the correct commands to run.

### How to Test (E2E open)
1. Run `yarn workspace @packages/launchpad cypress:open:e2e`
2. Shouldn't crash
3. Should standup the gql server under test (http://localhost:52100/graphql)
4. Should standup the dist'd server (http://localhost:5000/)
5. Can use launchpad to open browser in e2e mode
6. Tests succeed

### How to Test (CT open)
1. Run `yarn workspace @packages/launchpad cypress:open:ct`
2. Can use launchpad to open browser in ct mode
3. Tests succeed

### How to Test (CT + E2E Run mode)
1. Run `yarn workspace @packages/launchpad cypress:run`

### Acceptance Criteria

* Should be able to dist the launchpad frontend in production
* Should hit a specific gql endpoint for test mode
* Should host the distd production frontend via a webserver
* Should be able to open the dist'd launchpad in a normal browser
* Should wait for the prod build and gql servers to start up before running cypress.
